### PR TITLE
Extract substitution code out of v1alpha1 ✂

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This part doesn't really have anything to do with the API so, moving
it out of the v1alpha1 api package.

This is part of the "cleaning v1alpha1 and create v1alpha2" work :wink: 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

File: pkg/substitution/substitution_test.go [hunk 1]
```diff
@@ -15,13 +15,13 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1_test
+package substitution_test
```
Comment (bobcatfish) [inline pkg/substitution/substitution_test.go]: total nitpick but i think we might be able to get away with just `substitution` now? afaik `v1alpha_test` had to be used b/c we were having some kinda like conflicts or circular dependency problems or something (might be thinking of a different issue tho - i know this happens with the builder libs a lot! - but probably wont once you're done!!)

**Comment by vdemeester** [comment]
arf, another rebase :man_facepalming:

File: pkg/substitution/substitution_test.go (hunk 1 above)
Comment (vdemeester) [reply pkg/substitution/substitution_test.go]: I think `v1alpha_test` might have been for that reason indeed. I tend to like using `_test` package as it force the test to only use what the user can use and how he will use it :stuck_out_tongue:

Comment (bobcatfish) [reply pkg/substitution/substitution_test.go]: ah kk! makes sense to me. ill have to noodle on it a bit and maybe ill switch to the same thing!

**Pull request merged**